### ISSUE TITLE
Remote App quits with "error 1359" after few seconds

### ISSUE THREAD

**Issue by flops** [FreeRDP/FreeRDP#4398]
OS:
Arch linux.
OpenH264 1.7.0
Window 10 Enterprise

When starting app with:
```
xfreerdp +clipboard\
 /u:login /p:pass\
 /v:172.19.0.2\
 /app:"C:\Program Files\Internet Explorer\iexplore.exe"
```

The process is closing with log:
```
[02:19:09:971] [15978:15979] [INFO][com.freerdp.client.common.cmdline] - loading channelEx cliprdr
[02:19:09:971] [15978:15979] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rail
[02:19:09:992] [15978:15979] [ERROR][com.winpr.timezone] - Unable to get current timezone rule
[02:19:09:060] [15978:15979] [INFO][com.freerdp.gdi] - Local framebuffer format  PIXEL_FORMAT_BGRX32
[02:19:09:060] [15978:15979] [INFO][com.freerdp.gdi] - Remote framebuffer format PIXEL_FORMAT_RGB16
[02:19:09:064] [15978:15979] [INFO][com.winpr.clipboard] - initialized POSIX local file subsystem
[02:19:18:097] [15978:15990] [ERROR][com.freerdp.channels.rail.client] - context.ServerMinMaxInfo failed with error 1359
[02:19:18:097] [15978:15990] [ERROR][com.freerdp.channels.rail.client] - rail_order_recv failed with error 1359!
[02:19:18:098] [15978:15979] [ERROR][com.freerdp.core] - rail_virtual_channel_client_thread reported an error. Error was 1359
[02:19:18:098] [15978:15979] [INFO][com.freerdp.client.x11] - Network disconnect!
[02:19:18:098] [15978:15979] [ERROR][com.freerdp.client.x11] - Failed to check FreeRDP file descriptor
```
Build info:
```
This is FreeRDP version 2.0.0-dev2 (a90be5609)
Build configuration: BUILD_TESTING=OFF BUILTIN_CHANNELS=ON HAVE_AIO_H=1 HAVE_EVENTFD_H=1 HAVE_EXECINFO_H=1 HAVE_FCNTL_H=1 HAVE_INTTYPES_H=1 HAVE_JOURNALD_H=TRUE HAVE_MATH_C99_LONG_DOUBLE=1 HAVE_POLL_H=1 HAVE_PTHREAD_MUTEX_TIMEDLOCK=ON HAVE_PTHREAD_MUTEX_TIMEDLOCK_LIB=1 HAVE_PTHREAD_MUTEX_TIMEDLOCK_SYMBOL= HAVE_SYSLOG_H=1 HAVE_SYS_FILIO_H= HAVE_SYS_MODEM_H= HAVE_SYS_SELECT_H=1 HAVE_SYS_SOCKIO_H= HAVE_SYS_STRTIO_H= HAVE_TIMERFD_H=1 HAVE_TM_GMTOFF=1 HAVE_UNISTD_H=1 HAVE_XI_TOUCH_CLASS=1 WITH_ALSA=ON WITH_CCACHE=ON WITH_CHANNELS=ON WITH_CLIENT=ON WITH_CLIENT_AVAILABLE=1 WITH_CLIENT_CHANNELS=ON WITH_CLIENT_CHANNELS_AVAILABLE=1 WITH_CLIENT_COMMON=ON WITH_CLIENT_INTERFACE=OFF WITH_CUPS=OFF WITH_DEBUG_ALL=OFF WITH_DEBUG_CAPABILITIES=OFF WITH_DEBUG_CERTIFICATE=OFF WITH_DEBUG_CHANNELS=OFF WITH_DEBUG_CLIPRDR=OFF WITH_DEBUG_DVC=OFF WITH_DEBUG_KBD=OFF WITH_DEBUG_LICENSE=OFF WITH_DEBUG_MUTEX=OFF WITH_DEBUG_NEGO=OFF WITH_DEBUG_NLA=OFF WITH_DEBUG_NTLM=OFF WITH_DEBUG_RAIL=OFF WITH_DEBUG_RDP=OFF WITH_DEBUG_RDPDR=OFF WITH_DEBUG_RDPEI=OFF WITH_DEBUG_REDIR=OFF WITH_DEBUG_RFX=OFF WITH_DEBUG_RINGBUFFER=OFF WITH_DEBUG_SCARD=OFF WITH_DEBUG_SND=OFF WITH_DEBUG_SVC=OFF WITH_DEBUG_SYMBOLS=OFF WITH_DEBUG_THREADS=OFF WITH_DEBUG_TIMEZONE=OFF WITH_DEBUG_TRANSPORT=OFF WITH_DEBUG_TSG=OFF WITH_DEBUG_TSMF=OFF WITH_DEBUG_WND=OFF WITH_DEBUG_X11=OFF WITH_DEBUG_X11_CLIPRDR=OFF WITH_DEBUG_X11_LOCAL_MOVESIZE=OFF WITH_DEBUG_XV=OFF WITH_DIRECTFB=OFF WITH_EVENTFD_READ_WRITE=1 WITH_FFMPEG=ON WITH_GFX_H264=ON WITH_GPROF=OFF WITH_GSM=OFF WITH_GSSAPI=OFF WITH_GSTREAMER_0_10=OFF WITH_GSTREAMER_1_0=ON WITH_ICU=OFF WITH_IPP=OFF WITH_JPEG=OFF WITH_LIBRARY_VERSIONING=ON WITH_LIBSYSTEMD=ON WITH_MACAUDIO=OFF WITH_MACAUDIO=OFF WITH_MACAUDIO_AVAILABLE=0 WITH_MANPAGES=ON WITH_MBEDTLS=OFF WITH_OPENH264=ON WITH_OPENSLES=OFF WITH_OPENSSL=ON WITH_OSS=ON WITH_PAM=ON WITH_PCSC=OFF WITH_PROFILER=OFF WITH_PULSE=ON WITH_SAMPLE=OFF WITH_SANITIZE_ADDRESS=OFF WITH_SANITIZE_ADDRESS_AVAILABLE=1 WITH_SANITIZE_LEAK=OFF WITH_SANITIZE_LEAK_AVAILABLE=1 WITH_SERVER=ON WITH_SERVER_CHANNELS=ON WITH_SERVER_INTERFACE=ON WITH_SMARTCARD_INSPECT=OFF WITH_SSE2=ON WITH_THIRD_PARTY=OFF WITH_VALGRIND_MEMCHECK=OFF WITH_VALGRIND_MEMCHECK_AVAILABLE=1 WITH_WAYLAND=ON WITH_X11=ON WITH_X264=OFF WITH_XCURSOR=ON WITH_XDAMAGE=ON WITH_XEXT=ON WITH_XFIXES=ON WITH_XI=ON WITH_XINERAMA=ON WITH_XKBFILE=ON WITH_XRANDR=ON WITH_XRENDER=ON WITH_XSHM=ON WITH_XTEST=ON WITH_XV=ON WITH_ZLIB=ON
Build type:          Release
CFLAGS:              -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -fPIC -Wall -Wno-unused-result -Wno-unused-but-set-variable -Wno-deprecated-declarations -fvisibility=hidden -Wimplicit-function-declaration -Wredundant-decls
Compiler:            GNU, 7.3.0
Target architecture: x64
```

Build flags:
```
-DCMAKE_INSTALL_PREFIX=/usr \
-DCMAKE_INSTALL_LIBDIR=lib \
-DWITH_PULSE=ON \
-DWITH_SERVER=ON \
-DWITH_OPENH264=ON \
```

**Comment by akallabeth** [comment]
Does it work with #4322 applied?

**Comment by chrislong** [comment]
I've encountered what I think is the same problem. It happens exactly when I click on a menu to open it.

I checked out 6950d95 (from akallabeth/FreeRDP.git) and now menus work. However, after a few minutes, it crashed with this error:

> X Error of failed request:  BadWindow (invalid Window parameter)
>   Major opcode of failed request:  18 (X_ChangeProperty)
>   Resource id in failed request:  0x0
>   Serial number of failed request:  46398
>   Current serial number in output stream:  46409

I think opening a menu (that worked previously) caused it, but I'm not sure. I cannot reproduce it.

**Comment by akallabeth** [comment]
@chrislong please check out the pull request #4322, the commit you mention is only half of it

**Comment by chrislong** [comment]
@akallabeth Thanks for the tip. I checked out PR #4322 (with `git fetch origin pull/4322/head:localbranch` and `git checkout localbranch`) and built in a new tree with the default configuration. I can still reproduce the error if I repeatedly click on a menu to open and close it. `./xfreerdp --version` reports:

`This is FreeRDP version 2.0.0-dev2 (47110a723)`

If it matters, I'm running on Ubuntu 16.04 logging into Windows 10.

**Comment by chrislong** [comment]
I tried again with version
`This is FreeRDP version 2.0.0-dev2 (47110a723)`

I got it to crash with this command line and output (connecting to a Windows 10 VM):

```./xfreerdp /u:test /p:testpassword /v:windows10host /cert-name:mycert.cer /app:'c:\windows\notepad.exe' 
[21:17:15:169] [30851:30852] [INFO][com.freerdp.client.common.cmdline] - loading channelEx cliprdr
[21:17:15:169] [30851:30852] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rail
[21:17:15:797] [30851:30852] [ERROR][com.winpr.sspi.Kerberos] - error while getting credentials
[21:17:15:797] [30851:30852] [ERROR][com.winpr.sspi.Kerberos] - Kerberos credentials not found and could not be acquired
[21:17:15:797] [30851:30852] [WARN][com.winpr.negotiate] - No Kerberos credentials. Retry with NTLM
[21:17:15:797] [30851:30852] [WARN][com.winpr.sspi] - InitializeSecurityContextA status SEC_E_NO_CREDENTIALS [0x8009030E]
[21:17:16:262] [30851:30852] [INFO][com.freerdp.gdi] - Local framebuffer format  PIXEL_FORMAT_BGRX32
[21:17:16:262] [30851:30852] [INFO][com.freerdp.gdi] - Remote framebuffer format PIXEL_FORMAT_RGB16
[21:17:16:352] [30851:30852] [INFO][com.winpr.clipboard] - initialized POSIX local file subsystem
[21:17:18:532] [30851:30852] [INFO][com.freerdp.client.x11] - Logon Error Info LOGON_FAILED_OTHER [LOGON_MSG_SESSION_CONTINUE]
X Error of failed request:  BadWindow (invalid Window parameter)
  Major opcode of failed request:  18 (X_ChangeProperty)
  Resource id in failed request:  0x0
  Serial number of failed request:  46196
  Current serial number in output stream:  46202
```

Before that, I tried with another machine running Windows Server 2016 and could not make that crash. It could be the version of Windows or maybe it's that the crash occurs when running against a local VM but not one running over the internet (on AWS EC2).

**Comment by ShaunMaher** [comment]
Hi all.

I'm not 100% sure I'm getting the same issue or a slightly different one.

I have built FreeRDP from the current Git master (a9ecd6a at time of writing) and applied all of the changes that make up #4322.  I manually applied each of the changes in "Fixed #3676, #4269: Screen update for RAIL with GFX" as it no longer applies automatically and I'm not good enough with git to have it fix it for me.

Anyway, when using rail ("xfreerdp /u:DOMAIN\\username /v:<ip address> /app:"||visio" +clipboard +fonts /drive:home,/home/username', Windows Server 2016) I get the following error intermittently:
```
X Error of failed request:  BadWindow (invalid Window parameter)
  Major opcode of failed request:  20 (X_GetProperty)
  Resource id in failed request:  0x46003c3
  Serial number of failed request:  8994
  Current serial number in output stream:  8994
```

I am able to connect and operate for a few minutes before it happens.  I think it is popup menus and/or tool tip like windows that are causing it, but Visio is always popping up some crap so it's hard to be sure.

I use this daily for work so I'm more than happy to give any proposed fixes a good workout.

Cheers.
Shaun.
